openedx_rocketchat/static/js/rocket_chat_discussion.js

```diff
@@ -0,0 +1,75 @@
+
+class RocketChatNavigation {
+
+    constructor(url_change_role) {
+        this.pageContentMain = $(".page-content-main");
+        this.pageContentSecondary = $(".page-content-secondary");
+        this.url_change_role = url_change_role;
+
+        this.loadComponents(this.pageContentMain, this.pageContentSecondary);
+
+        $(window).bind('hashchange', $.proxy(function(){
+            this.loadComponents(this.pageContentMain, this.pageContentSecondary);
+        },this));
+
+        $(".change-role-rocketchat").unbind().click($.proxy(function() {
+            this.change_role(this.url_change_role);
+        }, this))
+    }
+
+
+    change_role(url_change_role){
+        var input = $(".rocket-chat-user-input")
+        var username = input.val();
+        var role = "coach";
+        $.ajax({
+            type: "POST",
+            url: url_change_role,
+            data: {"username": username, "role": role},
+            success: function(){
+                input.val("");
+                alert("The role has been changed");
+            },
+            error: function (request, status, error) {
+                alert(request.responseText);
+            },
+        });
+    }
+
+
+    loadComponents(pageContentMain, pageContentSecondary){
+
+        var navigationOptions = $(".rocket-chat-options").children();
+
+        if(window.location.hash == "#main"){
+            pageContentSecondary.hide();
+            $(navigationOptions[1]).removeClass("active");
+            pageContentMain.fadeIn("slow");
+            $(navigationOptions[0]).addClass("active");
+
+        }else if (window.location.hash == "#settings"){
```

Comment:
You need to ensure that no other component has the #settings id, is better to rename to another more accurate name. @andrey-canon

Reply:
This is not an id, this is a hash sign in the url

Reply:
So, if i change manually the url, i think this it will not work property, right? @andrey-canon

Reply:
if you change that manually and press enter, you will redirect to the respective view.